src/herder/TransactionQueue.cpp

```diff
@@ -703,11 +705,22 @@ TransactionQueue::getMaxOpsToFloodPerPeriod() const
 
     if (mApp.getConfig().FLOOD_TX_PERIOD_MS != 0)
     {
-        opsToFlood = bigDivide(opsToFlood, cfg.FLOOD_TX_PERIOD_MS,
-                               cfg.getExpectedLedgerCloseTime().count() * 1000,
-                               Rounding::ROUND_UP);
+        // perform a linear interpolation of how many ops we should have flooded
```

Comment:
I don't think this is what we want to do. Imagine the following scenario:

1. `FLOOD_TX_PERIOD_MS = 1000`, `FLOOD_OP_RATE_PER_LEDGER = 2000`
2. At T=0 a ledger closes
3. At T=3.9s a burst of 1000 transactions are submitted
4. At T=4s the broadcast timer triggers and all 1000 transactions flood

Basically, this method says that a burst that happens early in a ledger will be treated differently from a burst that happens late in a ledger. I think that all bursts should be throttled equally regardless of when they arrive.

Reply:
As an additional small advantage, reverting this change simplifies the entire transaction queue by eliminating `mBroadcastSteps` and `mBroadcastOpCount`.

Reply:
I ended up with this because of rounding issues: with stateless throttling, we end up pretty far from the goal because the math doesn't work out well.

For example testnet is configured with 105 ops/ledger, with any "sane" flood_tx_period_ms (`FLOOD_TX_PERIOD_MS=1000`), and `FLOOD_OP_RATE_PER_LEDGER=105*2`, we end up with `105*2*1000/(5*1000) = 42`.

We can't flood transactions with more than 42 operations. If we round up to the next multiplier of 100, we overshoot by more than a factor of 2.

There is a similar problem with larger ledger sizes, because we probably want to use periods that are in the order of 100-500ms to avoid re-creating spikes: `FLOOD_TX_PERIOD_MS = 100`,  `FLOOD_OP_RATE_PER_LEDGER = 2000` which yields `2000*100/(5*1000)=40` so we can't flood transactions with more than 40 ops.

Note that the current code was indeed not finished - I have `mBroadcastOpCount` in there that is not used. I really needed a vacation :)

Maybe all we need to do is instead accrue each period a "broadcast balance", but cap the balance carried over to the next period to 99 ops. There would not be any counter anymore, just that balance:
```
    size_t opsToFlood;
    if (mApp.getConfig().FLOOD_TX_PERIOD_MS != 0)
    {
// opsPerPeriod initialized with bigDivide(opsToFloodLedger, cfg.FLOOD_TX_PERIOD_MS, cfg.getExpectedLedgerCloseTime().count() * 1000, Rounding::ROUND_UP)
        opsToFlood = mBroadcastOpCount + opsPerPeriod;
    }
    else
    {
        // else, flood the target at once
        opsToFlood = opsToFloodLedger;
    }
```

and in `broadcastSome`:
```
...
    // carry over remainder, up to 99 ops
    mBroadcastOpCount = std::min<size_t>(opsToFlood, 99);
    return totalOpsToFlood != 0;
}
```

What do you think?

Reply:
I added a commit with the actual change

Reply:
This is a good solution. Simple and solves the problem.